logs: add logsource tag and update e2e expectations

gouri-yerra opened the pull request:
### What does this PR do?
Adds a per-line tag showing the container output stream: logsource:stdout or logsource:stderr

Implemented in:
- pkg/logs/message/message.go (new helper LogSourceTag)
- pkg/logs/internal/parsers/kubernetes/kubernetes.go (tag from stream token)
- pkg/logs/internal/parsers/dockerfile/docker_file.go (tag from JSON stream)
- pkg/logs/internal/parsers/dockerstream/docker_stream.go (derive from header byte 1/2)

Also added releasenotes/notes/logsource-tag-stderr.yaml

### Motivation
- Users want to route/filter/alert on stderr vs stdout without parsing rules.
- We already parse the stream; surfacing it as a tag is low-risk and improves UX.
- Related: https://github.com/DataDog/datadog-agent/issues/33375 (internal: [AGNTLOG-387](https://datadoghq.atlassian.net/browse/AGNTLOG-387))

### Describe how you validated your changes
Andrew Q got my image from ci and made a file `docker-compose.yml` with the following content
```
services:
  agent:
    image: datadog/agent-dev:gyerra-feat-logsource-tag-stderr-stdout-py3-jmx
    environment:
      DD_API_KEY: "your api key"
      DD_LOGS_ENABLED: "true"
      DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL: "true"
      DD_LOGS_CONFIG_DOCKER_CONTAINER_USE_FILE: "false" # important on macOS Docker Desktop
      DD_LOG_LEVEL: "debug"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro

  logsource-test:
    image: busybox:1.36
    # docker compose interpolates `$VAR` from host env; escape `$` as `$$` so the container shell sees it.
    command:
      - sh
      - -c
      - |
        i=0
        while true; do
          echo "hello stdout $$i"
          echo "hello stderr $$i" 1>&2
          i=$$((i+1))
          sleep 1
        done
```
and after running `docker compose up -d` he was able to start both the containers up and run them and then he did `docker exec -it {container-id} bin/bash`  to get into the shell of the agent container and i was able to run `agent stream-logs` to see the logs with the tags
log:
<img width="2262" height="534" alt="image" src="https://github.com/user-attachments/assets/4f0eec6a-5bff-49b0-9616-6972ae70954d" />

### Additional Notes


[AGNTLOG-387]: https://datadoghq.atlassian.net/browse/AGNTLOG-387?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

gouri-yerra committed:
Tag stderr logs with LogSource
Trying to fix merge conflict

gouri-yerra committed:
adding whitespace for testing

gouri-yerra committed:
Trying again to commit changes but with proper permissions

gouri-yerra committed:
adding the 4 files that actually add the logsource tag

gouri-yerra committed:
Merge branch 'main' into gouri/AGNTLOG-387-logsource-tag-2

gouri-yerra committed:
trying to fix lint errors

gouri-yerra committed:
e2e: allow logsource:* as optional tags in testLog

gouri-yerra committed:
Merge branch 'main' into gouri/AGNTLOG-387-logsource-tag-2

gh123man commented on pkg/dyninst/integration_test.go:
```diff
@@ -220,7 +220,7 @@ func testDyninst(
 
 	// Trigger the function calls, receive the events, and wait for the process
 	// to exit.
-	t.Logf("Triggering function calls at %s", time.Now().Format(time.RFC3339))
```
The changes in this file seem unrelated to the PR. Can we revert these changes?

gouri-yerra replied:
Reverted!!

gouri-yerra committed:
Gouri/agntlog 387 logsource tag 2 (#44651)

### What does this PR do?

### Motivation

### Describe how you validated your changes

### Additional Notes

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: spencer.gilbert <spencer.gilbert@datadoghq.com>
Co-authored-by: Dustin Long <me@dustmop.io>
Co-authored-by: dustin.long <dustin.long@datadoghq.com>
Co-authored-by: Paul Cacheux <paul.cacheux@datadoghq.com>
Co-authored-by: temporal-staging-github-worker-1[bot] <83677184+temporal-staging-github-worker-1[bot]@users.noreply.github.com>
Co-authored-by: AliDatadog <125997632+AliDatadog@users.noreply.github.com>
Co-authored-by: Pierre-Louis Veyrenc <pierrelouis.veyrenc@datadoghq.com>
Co-authored-by: Hongshi Guo <hongshi.guo@datadoghq.com>
Co-authored-by: ilia.kurenkov <ilia.kurenkov@datadoghq.com>
Co-authored-by: Maxime Chambre <99009932+Enzu83@users.noreply.github.com>
Co-authored-by: maxime.chambre <maxime.chambre@datadoghq.com>
Co-authored-by: dd-octo-sts[bot] <200755185+dd-octo-sts[bot]@users.noreply.github.com>
Co-authored-by: FlorentClarret <florent.clarret@datadoghq.com>
Co-authored-by: kacper-murzyn <89013263+kacper-murzyn@users.noreply.github.com>
Co-authored-by: Márcio Kovags <marcio.kovags@datadoghq.com>
Co-authored-by: Pierre Gimalac <pierre.gimalac@datadoghq.com>
Co-authored-by: Dmitry <bitomaxsp@gmail.com>
Co-authored-by: dmitry.ponomarev <dmitry.ponomarev@datadoghq.com>
Co-authored-by: Kyle Neale <kyle.a.neale@gmail.com>
Co-authored-by: kyle.neale <kyle.neale@datadoghq.com>

gouri-yerra committed:
Merge branch 'main' into gouri/AGNTLOG-387-logsource-tag-2

gouri-yerra committed:
logs: gate logsource:{stdout,stderr} behind logs_config.add_logsource_tag (default false) (AGNTLOG-387)
Tryig to fix merge conflict
"logs: gate logsource:{stdout,stderr} behind logs_config.add_logsource_tag (default false) (AGNTLOG-387)"

gouri-yerra committed:
Update integration_test.go

Fixing time e2e testings

gouri-yerra committed:
Update kubernetes.go

Kubernetes parser: add tag when enabled
File: pkg/logs/internal/parsers/kubernetes/kubernetes.go
Import: pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
Conditional append:
if logs_config.add_logsource_tag and stream is stdout/stderr → append message.LogSourceTag(stream) to msg.ParsingExtra.Tags

gouri-yerra committed:
Update docker_file.go

Docker JSON-file parser: add tag when enabled
File: pkg/logs/internal/parsers/dockerfile/docker_file.go
Same conditional append using the stream field

gouri-yerra committed:
Update docker_stream.go

Docker stream (API) parser: derive stream from header and add tag when enabled
File: pkg/logs/internal/parsers/dockerstream/docker_stream.go
Determine stream from first header byte (1=stdout, 2=stderr) and conditionally append LogSourceTag

gouri-yerra committed:
Update kubernetes.go

only a single, flag‑guarded block.

gouri-yerra committed:
Trying to fix tag block
Fixing merge conflicts in parser file.

gouri-yerra committed:
Merge branch 'main' into gouri/AGNTLOG-387-logsource-tag-2

gouri-yerra committed:
logs: resolve dockerstream parser merge conflict

gouri-yerra committed:
logs: gofmt parsers

gouri-yerra committed:
Merge branch 'main' into gouri/AGNTLOG-387-logsource-tag-2

gouri-yerra committed:
Merge branch 'main' into gouri/AGNTLOG-387-logsource-tag-2

gouri-yerra committed:
Merge branch 'main' into gouri/AGNTLOG-387-logsource-tag-2

gouri-yerra committed:
Merge branch 'main' into gouri/AGNTLOG-387-logsource-tag-2

aiuto reviewed (approved):
For reasons in thread: https://dd.slack.com/archives/C08SK4B0FK8/p1767627713623179

I am approving force-merge for agent-build.
